Source/JavaScriptCore/jit/JITOperations.cpp

```diff
@@ -2045,7 +2045,7 @@ JSC_DEFINE_JIT_OPERATION(operationCallDirectEvalStrict, EncodedJSValue, (void* f
     CallFrame* calleeFrame = bitwise_cast<CallFrame*>(frame);
     calleeFrame->setCodeBlock(nullptr);
 
-    if (!isHostFunction(calleeFrame->guaranteedJSValueCallee(), globalFuncEval))
+    if (calleeFrame->guaranteedJSValueCallee() != calleeFrame->callerFrame()->codeBlock()->globalObject()->evalFunction())
```

Comment:
Is it correct even if the Eval is inside inlined CodeBlock from different realm's DFG CodeBlock?

Reply:
Oh, `Interpreter::eval()` resolved global object in way more complicated way than `calleeFrame->callerFrame()->codeBlock()->globalObject()`!!

I moved the check there.